### PR TITLE
[Logistic Regression] Support benchmarking in run_benchmark.sh

### PR DESCRIPTION
**Pull request by lijinf2**

### CODE REVIEW

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/core.py [hunk 1]
```diff
@@ -1198,8 +1198,10 @@ def _out_schema(self, input_schema: StructType) -> Union[StructType, str]:
 
         schema = f"{pred.prediction} {pyspark_type}"
         if self._has_probability_col():
+            print("got here: has prob")
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/core.py:1201]: Delete

File: python/src/spark_rapids_ml/core.py [hunk 2]
```diff
@@ -1198,8 +1198,10 @@ def _out_schema(self, input_schema: StructType) -> Union[StructType, str]:
...
             schema = f"{schema}, {pred.probability} array<{pyspark_type}>"
         else:
+            print("got here: no prob")
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/core.py:1204]: Delete

File: python/benchmark/benchmark/bench_logistic_regression.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            # train_df = scaler.fit(train_df).transform(train_df).drop(features_col,"features_col_vec").withColumn(features_col, vector_to_array("scaled_col_vec","float32"))
+            # train_df.cache()
+        else:
+            from pyspark.ml.classification import (
+                LogisticRegression as SparkLogisticRegression,
+            )
+
+            lr = SparkLogisticRegression(**params)  # type: ignore[assignment]
+            benchmark_string = "Spark ML LogisticRegression training"
+            # scaler = StandardScaler(withMean=True, withStd=True, inputCol=features_col, outputCol="scaled_col_vec")
```
Comment (eordentlich) [inline python/benchmark/benchmark/bench_logistic_regression.py:83]: Similar

**Comment by lijinf2** [comment]
build

**Review by eordentlich — APPROVED** [review]
👍

File: python/benchmark/benchmark/bench_logistic_regression.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+        model, fit_time = with_benchmark(benchmark_string, lambda: lr.fit(train_df))
+
+        # placeholder try block till hasSummary is supported in gpu model
+        try:
+            if model.hasSummary:
+                print(f"total iterations: {model.summary.totalIterations}")
+                print(f"objective history: {model.summary.objectiveHistory}")
+        except:
+            print("model does not have hasSummary attribute")
```
Comment (leewyang) [inline python/benchmark/benchmark/bench_logistic_regression.py:90]: Probably would be better to implement a `hasSummary()` method in the model that just returns `False` for now.

File: python/run_benchmark.sh [hunk 5]
```diff
@@ -66,7 +67,7 @@ knn_num_rows=$num_rows
 num_cols=${num_cols:-3000}
 
 # for large num_rows (e.g. > 100k), set below to ./benchmark/gen_data_distributed.py and /tmp/distributed
-gen_data_script=./benchmark/gen_data.py
+gen_data_script=${gen_data_script:-./benchmark/gen_data.py}
```
Comment (leewyang) [inline python/run_benchmark.sh:70]: Is there a different gen_data script that should be included in this PR?  Or is this just a hook for debugging?

**Comment by leewyang** [comment]
Sorry for the late review...

File: python/benchmark/benchmark/bench_logistic_regression.py (hunk 4 above)
Comment (lijinf2) [reply python/benchmark/benchmark/bench_logistic_regression.py:90]: Revised in a new PR: https://github.com/NVIDIA/spark-rapids-ml/pull/375

**Pull request merged**